var/spack/repos/builtin/packages/openscenegraph/package.py

```diff
@@ -33,8 +33,8 @@ class Openscenegraph(CMakePackage):
     )  # Qt windowing system was moved into separate osgQt project
     depends_on("qt@4:", when="@3.2:3.5.4")
     depends_on("qt@:4", when="@:3.1")
-    depends_on("libxinerama")
-    depends_on("libxrandr")
+    depends_on("libxinerama", when="platform=linux")
+    depends_on("libxrandr", when="platform=linux")
```

Comment:
I think we still support `cray` as a separate "platform", although it is linux. You'd have to duplicate those lines.